packages/ui/src/components/form-input/components/form-text-input.tsx

```diff
@@ -36,13 +22,10 @@ const FormTextInput = forwardRef<HTMLInputElement, FormTextInputPropsType>((prop
       name={props.name}
       control={formContext.control}
       render={({ field, fieldState }) => {
-        // Use proper React Hook Form reference handling
-        field.ref = setRefs
-
         // form error takes precedence over props.error
-        const errorMessage = fieldState.error?.message || props.error
+        const error = fieldState.error?.message ?? props.error
 
-        return <TextInput {...props} {...field} error={errorMessage} />
+        return <TextInput {...{ ...props, ...field, error }} ref={ref} theme={error ? 'danger' : 'default'} />
```

Comment:
Not a big fan of spreading into a spread. It takes a bit more thinking than just using props when possible.
```suggestion
        return <TextInput {...props} {...field} error={error} ref={ref} />
```

Reply:
+1

I felt the current version is more readable than this. It looks bit complicated.

Reply:
Fixed